DashAI/back/api/api_v1/endpoints/runner.py

```diff
@@ -0,0 +1,99 @@
+import logging
+from typing import Dict
+
+from fastapi import APIRouter, Depends, Response, status
+from fastapi.exceptions import HTTPException
+from sqlalchemy import exc
+from sqlalchemy.orm import Session
+
+from DashAI.back.api.api_v1.endpoints.components import _intersect_component_lists
+from DashAI.back.api.deps import get_db
+from DashAI.back.core.config import component_registry, settings
+from DashAI.back.database.models import Dataset, Experiment, Run
+from DashAI.back.dataloaders.classes.dashai_dataset import DashAIDataset, load_dataset
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@router.post("/")
+async def execute_run(run_id: int, db: Session = Depends(get_db)):
+    """
+    Train and evaluate the given run.
+
+    Parameters
+    ----------
+    run_id : int
+        id of the run to query, train and evaluate.
+
+    """
+    try:
+        run: Run = db.get(Run, run_id)
+        if not run:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail="Run not found"
+            )
+        exp: Experiment = db.get(Experiment, run.experiment_id)
+        if not exp:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail="Experiment not found"
+            )
+        dat: Dataset = db.get(Dataset, exp.dataset_id)
+        if not dat:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail="Dataset not found"
+            )
+    except exc.SQLAlchemyError as e:
+        log.exception(e)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Internal database error",
+        )
+    run_instance = component_registry[run.model_name]["class"](**run.parameters)
```

Comment:
This line of code is too complex: it has too many operations.
Split at least in two:

```suggestion
    # Instance the model with the run parameters.
    run_model_class = component_registry[run.model_name]["class"]
    run_instance = run_model_class(**run.parameters)
```